webapp/src/views/projects/translations/TranslationTools/useTranslationTools.ts

```diff
@@ -0,0 +1,76 @@
+import { useMemo, useRef } from 'react';
+import { useApiQuery } from 'tg.service/http/useQueryApi';
+
+type Props = {
+  projectId: number;
+  keyId: number;
+  targetLanguageId: number;
+  baseText?: string;
+  enabled?: boolean;
+  onValueUpdate: (value: string) => void;
+};
+
+export const useTranslationTools = ({
+  projectId,
+  keyId,
+  baseText,
+  targetLanguageId,
+  onValueUpdate,
+  enabled = true,
+}: Props) => {
+  const memory = useApiQuery({
+    url: '/v2/projects/{projectId}/suggest/translation-memory',
+    method: 'post',
+    // @ts-ignore
+    query: { keyId, targetLanguageId, baseText },
```

Comment:
Isn't there something wrong in the type definitions? Why are you ts-ignoring in cintext of out internal code?

Reply:
I'll add comment. I need to add keyId to query for react-query to reset the data when key changes.